docs/hub/timm.md

```diff
@@ -0,0 +1,162 @@
+</div>
+<div class="flex justify-center">
+<img class="block dark:hidden" src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/hub/libraries-timm_snippet2.png"/>
+<img class="hidden dark:block" src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/hub/libraries-timm_snippet2-dark.png"/>
+</div>
+
+### Inference
+
+The snippet below shows how you can perform inference on a `timm` model loaded from the Hub:
+
+```py
+import timm
+import torch
+from PIL import Image
+from timm.data import resolve_data_config
+from timm.data.transforms_factory import create_transform
+
+# Load from Hub 🔥
+model = timm.create_model(
+    'hf_hub:nateraw/resnet50-oxford-iiit-pet',
+    pretrained=True
+).eval()
+
+# Create Transform
+transform = create_transform(**resolve_data_config(model.pretrained_cfg, model=model))
+
+# Get the labels from the model config
+labels = model.pretrained_cfg['labels']
+top_k = min(len(labels), 5)
+
+# Use your own image file here...
+image = Image.open('boxer.jpg').convert('RGB')
+
+# Process PIL image with transforms and add a batch dimension
+x = transform(image).unsqueeze(0)
+
+# Pass inputs to model forward function to get outputs
+out = model(x)
+
+# Apply softmax to get predicted probabilities for each class
+probabilities = torch.nn.functional.softmax(out[0], dim=0)
+
+# Grab the values and indices of top 5 predicted classes
+values, indices = torch.topk(probabilities, top_k)
+
+# Prepare a nice dict of top k predictions
+predictions = [
+    {"label": labels[i], "score": v.item()}
+    for i, v in zip(indices, values)
+]
+print(predictions)
+```
+
+This should leave you with a list of predictions, like this:
+
+```py
+[
+    {'label': 'american_pit_bull_terrier', 'score': 0.9999998807907104},
+    {'label': 'staffordshire_bull_terrier', 'score': 1.0000000149011612e-07},
+    {'label': 'miniature_pinscher', 'score': 1.0000000149011612e-07},
+    {'label': 'chihuahua', 'score': 1.0000000149011612e-07},
+    {'label': 'beagle', 'score': 1.0000000149011612e-07}
+]
+```
+
+## Sharing your models
+
+You can share your `timm` models directly to the Hugging Face Hub. This will publish a new version of your model to the Hugging Face Hub, creating a model repo for you if it doesn't already exist.
+
+Before pushing a model, make sure that you've logged in to Hugging Face:
+
+```sh
+python -m pip install huggingface_hub
+huggingface-cli login
+```
+
+Alternatively, if you prefer working from a Jupyter or Colaboratory notebook, once you've installed `huggingface_hub` you can log in with:
+
+```py
+from huggingface_hub import notebook_login
+notebook_login()
+```
+
+Then, push your model using the `push_to_hf_hub` method:
+
+```py
+import timm
+
+# Build or load a model, e.g. timm's pretrained resnet18
+model = timm.create_model('resnet18', pretrained=True, num_classes=4)
```

Comment:
No `hf_hub` prefix here because this model isn't being loaded from the Hub?

Reply:
Yup, it's just loading it from wherever timm's default config gets it from. In this case it's from the PyTorch website (https://github.com/rwightman/pytorch-image-models/blob/1b278136c3f6c5f573e8007ff0fb0caffc8fc6c3/timm/models/resnet.py#L48).

Reply:
correct!